### ISSUE TITLE
Pubmine ads don't work 1905211840370

### ISSUE THREAD

**Issue by shackijj** [ampproject/amphtml#22429]
## What's the issue?

Pubmine ads don't work.

## How do we reproduce the issue?

Open a page containing a pubmine ad.

The problem can't be reproduced using instructions described [here](https://github.com/ampproject/amphtml/blob/master/ads/README.md#verify-your-examples). Therefore it wasn't noticed during QA and code review.

## What browsers are affected?

All browsers

## Which AMP version is affected?

1905211840370
1905140117570/production

#22426

**Comment by prateekbh** [comment]
do you have a sample page for repro?

**Comment by shackijj** [comment]
Yes, [here](https://theconservativetreehouse.com/2019/05/22/may-22nd-2019-presidential-politics-trump-administration-day-853/amp/) it is.

**Comment by prateekbh** [comment]
But this is broken even in 1905140117570/production as well in, right?
You'll need a cherry-pick to production and canary.

**Comment by calebcordry** [comment]
Were you able to determine what context is actually making this break? I saw your broken test page, but even when I run that page through my local proxy it still works.

**Comment by shackijj** [comment]
It's related to 1905140117570

**Comment by shackijj** [comment]
@calebcordry The page works. The problem is that ads on the page are broken. There are errors in console.

**Comment by calebcordry** [comment]
@shackijj I understand we are looking at the ads here. I am just trying to understand the cases where the passed in global != the actual global. It's not something I need you to solve now, just thought you might have uncovered in your debugging.

**Comment by shackijj** [comment]
> @shackijj I understand we are looking at the ads here. I am just trying to understand the cases where the passed in global != the actual global. It's not something I need you to solve now, just thought you might have uncovered in your debugging.

I think that the problem is caused by advanced optimization applied by google-closure-compiler for production builds. The code works fine when it's not minimized. I wonder why the compiler or linter doesn't throw errors or warnings regarding using an undefined variable.

When I implemented the feature I used [these](https://github.com/ampproject/amphtml/blob/master/ads/README.md#verify-your-examples) instructions to make sure that everything is fine. The ads still work on the test page `http://localhost:8000/examples/ads.amp.html?type=pubmine` in master branch. While debuging the issue, I noticed that `gulp` command produces unminified build.  I spent a couple of hours trying to figure out how to produce minified builds using gulp tasks for verifying examples in /ads directory, but I didn't succeed.
